aws_lambda_powertools/utilities/parser/parser.py

```diff
@@ -0,0 +1,156 @@
+import logging
+from typing import Any, Callable, Dict, Optional
+
+from pydantic import BaseModel, ValidationError
+
+from ...middleware_factory import lambda_handler_decorator
+from ..typing import LambdaContext
+from .envelopes.base import BaseEnvelope
+from .exceptions import InvalidEnvelopeError, InvalidSchemaTypeError, SchemaValidationError
+
+logger = logging.getLogger(__name__)
+
+
+@lambda_handler_decorator
+def parser(
```

Comment:
How would you feel about renaming this to `event_parser`? That makes it less likely for people to accidentally do `from aws_lambda_powertools.utilities import parser` when they mean `from aws_lambda_powertools.utilities.parser import parser`